src/App.tsx

```diff
@@ -1,26 +1,181 @@
+  }, [todos, newTodoInput]);
+
+  useEffect(() => {
+    if (deleteTodoId) {
+      setCurrentError(null);
+
+      deleteTodo(deleteTodoId)
+        .then(() => {
+          getTodos()
+            .then(setTodos)
+            .catch(() => {
+              setCurrentError(Error.CannotLoad);
+            });
+        })
+        .catch(() => {
+          setCurrentError(Error.CannotDelete);
+        })
+        .finally(() => setDeleteTodoId(null));
+    }
+  }, [deleteTodoId]);
+
+  const deleteTodoById = useCallback((id: number) => {
+    setDeleteTodoId(id);
+
+    deleteTodo(id)
+      .then(() =>
+        setTodos((currentTodos: Todo[]) =>
+          currentTodos.filter((todo: Todo) => todo.id !== id),
+        ),
+      )
+      .catch(() => {
+        setCurrentError(Error.CannotDelete);
+      })
+      .finally(() => setDeleteTodoId(null));
+  }, []);
+
+  const createTodo = useCallback(() => {
+    const todoTitle = addTodoTitle?.trim();
+
+    if (!todoTitle.length) {
+      setCurrentError(Error.EmptyTitle);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: addTodoTitle,
+      completed: false,
+    });
+
+    setIsNewTodoLoading(true);
+
+    addTodo(todoTitle)
+      .then(newTodo => {
+        setTodos(
+          (currentTodos: Todo[]) => [...currentTodos, newTodo] as Todo[],
+        );
+
+        setAddTodoTitle('');
+      })
+      .catch(() => setCurrentError(Error.CannotAdd))
+      .finally(() => {
+        setTempTodo(null);
+        setIsNewTodoLoading(false);
+      });
+  }, [addTodoTitle]);
+
+  const clearCompletedTodos = useCallback(() => {
+    const completedTodos = todos.filter((todo: Todo) => todo.completed);
+
+    for (const todo of completedTodos) {
+      deleteTodoById(todo.id);
+    }
+  }, [deleteTodoById, todos]);
+
+  const handleFilterChange = useCallback((filter: Filter) => {
+    return () => setCurrentFilter(filter);
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          handleChangeTitle={setAddTodoTitle}
+          addTodoTitle={addTodoTitle}
+          createTodo={createTodo}
+          newTodoInput={newTodoInput}
+          isNewTodoLoading={isNewTodoLoading}
+        />
+
+        {todos.length > 0 && (
+          <TodoList
+            todos={getFilteredTodos(todos, currentFilter)}
+            handleDeleteTodo={deleteTodoById}
+          />
+        )}
+
+        {tempTodo && (
```

Comment:
If you use my advice on how to work with tempTodo => this chunk can be completely removed